NineChronicles.Headless/GraphTypes/TransactionHeadlessQuery.cs

```diff
@@ -312,6 +295,45 @@ public TransactionHeadlessQuery(StandaloneContext standaloneContext)
             );
         }
 
+        private static object? TxResult(StandaloneContext standaloneContext, TxId txId)
+        {
+            if (!(standaloneContext.BlockChain is BlockChain blockChain))
+            {
+                throw new ExecutionError(
+                    $"{nameof(StandaloneContext)}.{nameof(StandaloneContext.BlockChain)} was not set yet!");
+            }
+
+            if (!(standaloneContext.Store is IStore store))
+            {
+                throw new ExecutionError(
+                    $"{nameof(StandaloneContext)}.{nameof(StandaloneContext.Store)} was not set yet!");
+            }
+
+            if (!(store.GetFirstTxIdBlockHashIndex(txId) is { } txExecutedBlockHash))
+            {
+                return blockChain.GetStagedTransactionIds().Contains(txId)
+                    ? new TxResult(TxStatus.STAGING, null, null, null, null, null)
+                    : new TxResult(TxStatus.INVALID, null, null, null, null, null);
+            }
+
+            try
+            {
+                TxExecution execution = blockChain.GetTxExecution(txExecutedBlockHash, txId);
+                Block txExecutedBlock = blockChain[txExecutedBlockHash];
+                return new TxResult(
+                    execution.Fail ? TxStatus.FAILURE : TxStatus.SUCCESS,
+                    txExecutedBlock.Index,
+                    txExecutedBlock.Hash.ToString(),
+                    execution.InputState,
+                    execution.OutputState,
+                    execution.ExceptionNames);
+            }
+            catch (Exception)
+            {
+                return new TxResult(TxStatus.INVALID, null, null, null, null, null);
+            }
```

Comment:
When the exception will be raised? I think the exception type should be specified more (not just `Exception`). If not, it will make us hard to debug why the request failed.

Reply:
I misread the change, this part was already working like this. It would be nice to fix it separately in the next PR.